cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaAssemblyOperator.java

```diff
@@ -83,40 +87,151 @@ public KafkaAssemblyOperator(Vertx vertx, boolean isOpenShift,
+     * is created or updated. It brings information used from a call to the next one and can be
+     * enriched if the subsequent call needs more information.
+     */
+    private static class KafkaClusterDescription {
+
+        private final KafkaCluster kafka;
+        private final Service service;
+        private final Service headlessService;
+        private final ConfigMap metricsConfigMap;
+        private final StatefulSet statefulSet;
+        private final Secret clientsCASecret;
+        private final Secret clientsPublicKeySecret;
+        private final Secret brokersClientsSecret;
+        private final Secret brokersInternalSecret;
+        private ReconcileResult<StatefulSet> diffs;
+
+        KafkaClusterDescription(KafkaCluster kafka, Service service, Service headlessService,
+                                ConfigMap metricsConfigMap, StatefulSet statefulSet,
+                                Secret clientsCASecret, Secret clientsPublicKeySecret,
+                                Secret brokersClientsSecret, Secret brokersInternalSecret) {
+            this.kafka = kafka;
+            this.service = service;
+            this.headlessService = headlessService;
+            this.metricsConfigMap = metricsConfigMap;
+            this.statefulSet = statefulSet;
+            this.clientsCASecret = clientsCASecret;
+            this.clientsPublicKeySecret = clientsPublicKeySecret;
+            this.brokersClientsSecret = brokersClientsSecret;
+            this.brokersInternalSecret = brokersInternalSecret;
+        }
+
+        KafkaCluster kafka() {
+            return this.kafka;
+        }
+
+        Service service() {
+            return this.service;
+        }
+
+        Service headlessService() {
+            return this.headlessService;
+        }
+
+        ConfigMap metricsConfigMap() {
+            return this.metricsConfigMap;
+        }
+
+        StatefulSet statefulSet() {
+            return this.statefulSet;
+        }
+
+        Secret clientsCASecret() {
+            return this.clientsCASecret;
+        }
+
+        Secret clientsPublicKeySecret() {
+            return this.clientsPublicKeySecret;
+        }
+
+        Secret brokersClientsSecret() {
+            return this.brokersClientsSecret;
+        }
+
+        Secret brokersInternalSecret() {
+            return this.brokersInternalSecret;
+        }
+
+        ReconcileResult<StatefulSet> diffs() {
+            return this.diffs;
+        }
+
+        Future<KafkaClusterDescription> withDiff(Future<ReconcileResult<StatefulSet>> r) {
+            return r.map(rr -> {
+               this.diffs = rr;
+               return this;
+            });
+        }
+
+        Future<KafkaClusterDescription> withVoid(Future<?> r) {
+            return r.map(this);
+        }
+    }
+
+    private final Future<KafkaClusterDescription> getKafkaCluster(ConfigMap assemblyCm, List<Secret> assemblySecrets) {
```

Comment:
Its called `getKafkaCluster`. But it returns `KafkaClusterDescription`. I think you should rename it.